src/dcd/server/autocomplete/complete.d

```diff
@@ -558,8 +558,10 @@ void setCompletions(T)(ref AutocompleteResponse response,
 	DSymbol*[] symbols = getSymbolsByTokenChain(completionScope, tokens,
 		cursorPosition, completionType);
 
-	if (symbols.length == 0)
+	if (symbols.length == 0){
+		lookupUFCSForDotChaining(completionScope, tokens, cursorPosition, response);
```

Comment:
it seems if there is at least one auto complete result, UFCS chains aren't going to be resolved. I don't think this is a great user experience because when the user types `foo.bar().` they are likely also going to want to see what kind of UFCS things they can type already. If it's only showing the UFCS completion there once the user has typed enough, they probably already know which functions they are looking for.

Auto completion greatly helps newcomers to the language find out how the language and libraries work, accurately showing possible completions (and showing most or all things you can do) is quite important.

Auto completing UFCS is also useful for experienced users to validate what they typed is indeed correct before going through the compiler, but I think that's a less important use-case.

Reply:
Yeah right, I need to add more test scenarios to. thanks